src/sail_json_backend/json.ml

```diff
@@ -398,14 +398,97 @@ let remove_identity_funcs op =
       else op
   ) op identity_funcs
 
+let optional_operand k op =
+    (*
+        The current regex pattern targets a specific function name pattern 
+        that starts with "maybe_*", such as "maybe_vmask", "maybe_lmul_flag", and "maybe_ta_flag".
+        Since those kinds are the most prevalant ones as of now(May
+        2024).
+
+        TODO: come up with a more robust implementation, that doesn't involve
+        specifying function names explicitly.
+    *)
+
+  let pattern = Str.regexp_case_fold ("maybe[A-Za-z0-9_]+\\(" ^ "(" ^ op ^ ")" ^ "\\)") in
+  (*let pattern = Str.regexp ("maybe_[A-Za-z_]+\\(" ^ "(" ^ op ^ ")" ^ "\\)") in*)
+  match Hashtbl.find_opt assembly_clean k with
+  | Some op_list ->
+      let captured_operand =
+        List.find_map (fun s ->
+          if Str.string_match pattern s 0 then
+            let captured_group = Str.matched_group 0 s in
+            let optional_operand_parts = String.split_on_char '(' captured_group in
+            let captured_op = List.hd optional_operand_parts in
+            debug_print ("Captured operand: " ^ captured_op);
+            Some captured_op
+          else
+            None
+        ) op_list
+      in
+      (match captured_operand with
+      | Some op -> Some true, op
+      | None -> None, "")
+  | None -> None, ""
+
+
+(*
+    This examines zero-bit patterns like 0b0, 0b00, 0b000, etc.,
+    retrieved from the mappings Hashtbl to deduce the default operands.
+
+    See:
+    https://github.com/ThinkOpenly/sail-riscv/blob/46750eed2796799fcc4bb220af8ce45ae9fee76c/
+    model/riscv_insts_vext_vset.sail#L51
+*)
+let default_operand mappings key =
+  if Hashtbl.mem mappings key then
+    let values = Hashtbl.find_all mappings key in
+    let zero_bit_pattern s = Str.string_match (Str.regexp "0b0*$") s 0 in
```

Comment:
I think to match *zero-only* bit strings, the pattern should be `0b0+$`. Although the current pattern will never match anything with a `1`, it would match "0b", as unlikely as that may be.